src/lib/components/SystemThemeListener.svelte

```diff
@@ -0,0 +1,32 @@
+<script lang="ts">
+  import { onDestroy, onMount } from "svelte";
+  import { themeStore } from "$lib";
+  import { LOCALSTORAGE_THEME_KEY } from "$lib/utils/theme.utils.js";
+
+  export let nnsOnChange: undefined | ((isDarkMode: boolean) => void);
+
+  // Set up our MediaQueryList
+  const prefersDarkMode = window.matchMedia("(prefers-color-scheme: dark)");
+
+  // Update the store if OS preference changes
+  const updateThemeOnChange = () => {
+    if (nnsOnChange) {
+      nnsOnChange(prefersDarkMode.matches);
+    } else {
+      // Only reset to system theme if no specific preference has been selected
+      if (window.localStorage.getItem(LOCALSTORAGE_THEME_KEY) !== null) return;
```

Comment:
I wonder if this should find a place in `theme.utils`, so we continue avoiding references to storage in any components. WDYT?

Something like:

```
// component
const updateThemeOnChange = () => {
    if (isThemeSelected()) {
       return;
    } 
    
          themeStore.resetToSystemSettings();
    }
  };
  
  // theme.utils
  
  export const isThemeSelected = (): boolean =>  notEmptyString(window.localStorage.getItem(LOCALSTORAGE_THEME_KEY))
```